src/xcode/ENA/ENA/Source/Scenes/TraceLocations/TraceLocationsCoordinator.swift

```diff
@@ -115,9 +116,12 @@ class TraceLocationsCoordinator {
 				}
 			),
 			onDismiss: { [weak self] didConsent in
-				self?.infoScreenShown = didConsent
-				if !didConsent {
-					self?.parentNavigationController?.popViewController(animated: false)
+				guard let self = self else {
+					return
+				}
+				self.infoScreenShown = didConsent
+				if !alreadyDidConsentOnce && !didConsent {
```

Comment:
why is the `alreadyDidConsentOnce` variable moved outside the closure. I think this can be much more readable be done. A least a comment would clarify what's going on here.

Reply:
Aditionally: Doesn't this inverts the logic? (`didConsent` -> !`didConsent`)

Reply:
It is outside the closure because the value is changes as soon as this screen is presented. I need to store the bool value from before the presentation.